src/parser/mod.rs

```diff
@@ -13337,6 +13376,42 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parse an expression, optionally followed by ASC or DESC (used in ORDER BY)
+    pub fn parse_create_index_expr<OPS: OperatorClass>(
+        &mut self,
+    ) -> Result<IndexColumn, ParserError> {
+        let expr = self.parse_expr()?;
+
+        let operator_class: Option<OPS> = self.parse_one_of_keywords(OPS::KEYWORDS).map(Into::into);
+        let asc = self.parse_asc_desc();
+
+        let nulls_first = if self.parse_keywords(&[Keyword::NULLS, Keyword::FIRST]) {
+            Some(true)
+        } else if self.parse_keywords(&[Keyword::NULLS, Keyword::LAST]) {
+            Some(false)
+        } else {
+            None
+        };
+
+        let with_fill = if dialect_of!(self is ClickHouseDialect | GenericDialect)
+            && self.parse_keywords(&[Keyword::WITH, Keyword::FILL])
+        {
+            Some(self.parse_with_fill()?)
+        } else {
+            None
+        };
```

Comment:
since this part is identical to parse_order_by_expr, can we reuse the same code for both? e.g. split out to its own function